ssl/s3_lib.c

```diff
@@ -3714,8 +3714,10 @@ const SSL_CIPHER *ssl3_choose_cipher(SSL *s, STACK_OF(SSL_CIPHER) *clnt,
         allow = srvr;
     }
 
-    tls1_set_cert_validity(s);
-    ssl_set_masks(s);
+    if (!SSL_IS_TLS13(s)) {
+        tls1_set_cert_validity(s);
```

Comment:
The sigalgs was the only bit I found that I was pretty sure was relevant.
So, tls_choose_sigalg() covers that and this is safe, thanks.

Though, in looking at how tmp.peer_sigalgs gets set, I found myself looking at tls_parse_ctos_sig_algs() and wondering whether it is correct to only call tls1_save_sigalgs() if not resuming, as for TLS 1.3 these extensions should be renegotiated on each handshake, IIRC.